File: backend/internal/domain/entity/runner.go
package entity

import (
	"time"

	"github.com/google/uuid"
)

const (
	RunnerTypeAct      = "act"
	RunnerTypeOfficial = "official"

	RunnerStatusOnline  = "online"
	RunnerStatusOffline = "offline"
	RunnerStatusBusy    = "busy"
)

type Runner struct {
	ID             uuid.UUID
	OrganizationID uuid.UUID
	Name           string
	Labels         []string
	OS             string
	Arch           string
	RunnerType     string
	Status         string
	LastSeenAt     *time.Time
	Ephemeral      bool
	CreatedAt      time.Time
	UpdatedAt      time.Time
}

File: types/runner.ts
export interface Runner {
  id: string;
  name: string;
  status: "online" | "offline" | "busy";
  labels: string[];
  runner_type: "act" | "official";
  last_seen_at: string | null;
}

export interface RegistrationTokenResponse {
  token: string;
  expires_at: string;
}

export interface RunnerListResponse {
  runners: Runner[];
}

export interface RegisterRunnerRequest {
  registration_token: string;
  name: string;
  labels: string[];
  os: string;
  arch: string;
  runner_type: "act" | "official";
}

export interface WorkflowJobResponse {
  id: string;
  assigned_runner_id: string | null;
  runner_type: "act" | "official" | null;
}
